lib/elixir/lib/calendar.ex

```diff
@@ -99,6 +123,31 @@ defmodule Calendar do
   """
   @callback datetime_to_string(year, month, day, hour, minute, second, microsecond,
                                time_zone, zone_abbr, utc_offset, std_offset) :: String.t
+
+  @doc """
+  Converts the given datetime (with time zone)
+  into the internal Calendar.rata_die format.
+  """
+  @callback datetime_to_rata_die(year, month, day, hour, minute, second, microsecond,
```

Comment:
I believe the weirdness comes from those contradictories statements:

>  I would also consider timezone handling to be something that should be part of a calendar module, rather than DateTime.

> Most other calendars do not take timezones into account.

When I look at those, my conclusion is the opposite: calendars should not take timezone into accounts. At the same time I am aware that a Bhuddist time in UTC is likely useless although I don't seem to be mathematically incorrect.